examples/run_CPMG.py

```diff
@@ -31,91 +31,71 @@
-    adcOffset = config_dict["adc_offset"],
-    carrierFreq_MHz = config_dict["carrierFreq_MHz"],
-    nPoints = nPoints,
-    nEchoes = config_dict["nEchoes"],
-    p90_us = config_dict["p90_us"],
-    repetition_us = config_dict["repetition_us"],
-    tau_us = config_dict["tau_us"],
-    SW_kHz = config_dict["SW_kHz"],
-    pad_start_us = pad_start,
-    pad_end_us = pad_end,
-    ret_data = None,
-)
+data = generic(
+        ppg_list = [
+                ("marker","start",1),
+                ("phase_reset", 1),
+                ("delay_TTL", config_dict['deblank_us']),
+                ("pulse_TTL", config_dict['p90_us'], "ph_cyc", ph1_cyc),
+                ("delay", tau_us),
+                ("delay_TTL", config_dict['deblank_us']),
+                ("pulse_TTL", 2.0 * config_dict['p90_us'], "ph_cyc", ph2_cyc),
+                ("delay", config_dict['deadtime_us']),
+                ("acquire", acq),
+                ("delay", pad_end),
+                ("delay",1.0), #matching the jumpto delay
+                ("marker", "echo_label", (config_dict['nEchoes'] - 1)), 
+                ("delay_TTL", config_dict['deblank_us']),
+                ("pulse_TTL", 2.0 * config_dict['p90_us'], "ph_cyc",ph2_cyc),
+                ("delay", config_dict['deadtime_us']),
+                ("acquire", acq),
+                ("delay", pad_end),
+                ("jumpto", "echo_label"), 
+                ("delay", config_dict['repetition_us']),
+                ("jumpto","start")
+            ],
+        nScans = config_dict["nScans"],
+        indirect_idx = 0,
+        indirect_len = len(config_dict['nEchoes']),
+        adcOffset = config_dict["adc_offset"],
+        carrierFreq_MHz = config_dict["carrierFreq_MHz"],
+        nPoints = nPoints,
+        SW_kHz = config_dict["SW_kHz"],
+        ret_data = None,
+    )
 # }}}
 # {{{ chunk and save data
-if phase_cycling:
-    data.chunk("t", ["ph1", "echo", "t2"], [len(ph1_cyc), config_dict["nEchoes"], -1])
-    data.setaxis("ph1", ph1_cyc / 4)
-    data.setaxis("echo", r_[0 : config_dict["nEchoes"]])
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.squeeze()
-    data.set_units("t2", "s")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = data.C
-    for_plot.ft('t2',shift=True)
-    for_plot.ft(['ph1'], unitary = True)
-    fl.next('FTed data')
-    fl.image(for_plot.C.mean("nScans")
-    )
-else:
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.rename("t","t2")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = echo_data.C
-    for_plot.ft('t2',shift=True)
-    fl.next('FTed data')
-    fl.image(for_plot)
+data.chunk("t", ["ph_overall", "ph_diff", "tE", "t2"], [len(Nph_overall), len(Nph_diff), config_dict["nEchoes"], nPoints])
```

Comment:
The generic ppg creates an `nScans` dimension, and does not squeeze.  You also write the code above allowing for nScans.  So, you should allow for that here, *as well as* when you visualize.

Reply:
"allow for that here" to me means include an nScans in the chunking and mean for imaging.... so I did that but I'm not sure if that was what you meant?

Reply:
based on your next comment I think that's right